In [3]:
import numpy as np

In [4]:
x=np.genfromtxt(r"C:\Users\yuvin\Downloads\Sample_data.csv", delimiter=',' , skip_header=1)
x

array([[ 0., 25., 40.],
       [ 1., 30., 35.],
       [ 2., 28., 38.],
       [ 3., 32., 41.],
       [ 4., 26., 37.],
       [ 5., 29., 36.],
       [ 6., 27., 39.],
       [ 7., 33., 42.],
       [ 8., 31., 38.],
       [ 9., 28., 37.],
       [10., 30., 40.],
       [11., 29., 36.]])

# Measures of Central Tendency:
### 1.Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.


In [7]:
product_A=x[:,1]
product_A

array([25., 30., 28., 32., 26., 29., 27., 33., 31., 28., 30., 29.])

In [8]:
product_B=x[:,2]
product_B

array([40., 35., 38., 41., 37., 36., 39., 42., 38., 37., 40., 36.])

In [9]:
np.mean(product_A)

29.0

In [10]:
np.mean(product_B)

38.25

#### avereage monthly sales for Product A and Product B is 29.0 and 38.25
*  avereage monthly sales for Product A is greater than ProductB
* sales of B is better than that of A

# 2.Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.

In [11]:
np.median(product_A)

29.0

In [12]:
np.median(product_B)

38.0

* The median value is almost close to the mode
* The median of productA is 29.0
* The median of productB is 38.0

# 3.Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.

In [14]:
import statistics

Product_A = x[:,1]
Product_B = x[:,2]

a = statistics.mode(Product_A)
b = statistics.mode(Product_B)
print('Mode sale for productA is:',a)
print('Mode sale for productB is:',b)


Mode sale for productA is: 30.0
Mode sale for productB is: 40.0


* The mode represnets the most occuring values
* The mode for productA is 30.0
* The mode for productB is 40.0
* The prouctA used to sell most times the 30.0 sales whereas productB used to sell 40.0 products in a month 

# 4.Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.


In [15]:
Product_A = x[:,1]
Product_B = x[:,2]

weight=np.ones(12)

weighted_A = weight * Product_A
a = np.mean(weighted_A)

weighted_B = weight * Product_B
b = np.mean(weighted_B)

print('Weighted mean of the sales data of Product A :', a)
print('Weighted mean of the sales data of Product B :', b)

Weighted mean of the sales data of Product A : 29.0
Weighted mean of the sales data of Product B : 38.25


* We get the weighted mean of productA is the median of productA
* The mean of ProductB is same as the weighted mean of product B

# Measures of Dispersion:
# 1.Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.


In [20]:
max_product_A=np.max(x[:,1])
max_product_A

33.0

In [21]:
min_product_A=np.min(x[:,1])
min_product_A

25.0

In [31]:
max_product_B=np.max(x[:,2])
max_product_B

42.0

In [32]:
min_product_B=np.min(x[:,2])
min_product_B

35.0

In [35]:
range_A=max_product_A-min_product_A
print('The range for productA is:',range_A)

The range for productA is: 8.0


In [36]:
range_B=max_product_B-min_product_B
print('The range for productA is:',range_B)

The range for productA is: 7.0


# 2.Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product.


In [37]:
Product_A = x[:,1]
Product_B = x[:,2]


a = np.mean(Product_A)
variance_product_a = np.sum((Product_A - a)**2) / len(Product_A)

b = np.mean(Product_B)
variance_product_b = np.sum((Product_B - b)**2) / len(Product_B)

print("Variance of Sales for Product A:", variance_product_a)
print("Variance of Sales for Product B:", variance_product_b)

Variance of Sales for Product A: 5.166666666666667
Variance of Sales for Product B: 4.354166666666667


* The variance of a is greater than that of b
* The distance from the mean for a is greater ,It says that the product a has the least frequency so the sales of a is lesser than b

# 3.Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.



In [38]:
Product_A = x[:,1]
Product_B = x[:,2]

a = np.mean(Product_A)
standard_deviation_product_a = np.sqrt(np.sum((Product_A - a)**2) / len(Product_A))# The squareroot is trmed as sqrt

b = np.mean(Product_B)
standard_deviation_product_b = np.sqrt(np.sum((Product_B - b)**2) / (len(Product_B)))

print( standard_deviation_product_a)
print( standard_deviation_product_b)

2.273030282830976
2.0866640042581523


#  Relative Measures of Dispersion:
## 1.Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.


In [39]:

Product_A = x[:,1]

a = np.mean(Product_A)

Std = np.std(Product_A)

coeff=(Std/a)*100

print('Coefficient of variation for the sales data of Product A :',coeff)

Coefficient of variation for the sales data of Product A : 7.838035458037848


# 2. Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.

In [40]:
Product_B = x[:,2]

B = np.max(Product_B) - np.min(Product_B)

b = np.mean(Product_B)

relative_range_B = (B/b)*100

print('The relative range of B:',relative_range_B)

The relative range of B: 18.30065359477124


* The relative range is inversly  proportional to the mean of sales

# 3. Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.

In [41]:
Product_A = x[:,1]

a = np.mean(Product_A)

Std = np.std(Product_A)

rel_std = (Std / a)*100

print('relative standard deviation for Product A :',rel_std)

relative standard deviation for Product A : 7.838035458037848
